core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java

```diff
@@ -53,7 +51,7 @@ public class TaskManagerConfig
     private DataSize maxLocalExchangeBufferSize = DataSize.of(128, Unit.MEGABYTE);
     private DataSize maxIndexMemoryUsage = DataSize.of(64, Unit.MEGABYTE);
     private boolean shareIndexLoading;
-    private int maxWorkerThreads = Runtime.getRuntime().availableProcessors() * 2;
+    private ThreadCount maxWorkerThreads = ThreadCount.valueOf("2C");
```

Comment:
This is two separate changes

1. the main one per PR title: allowing to configure `n` in `n * number of cores` formula
2. how the "number of cores" is determined

I think the changes should go as separate things. The (1) is obvious, but the (2) is tricky and needs to be thoroughly code commented